Sources/SwiftDriver/Jobs/GenericUnixToolchain+LinkerSupport.swift

```diff
@@ -147,14 +147,24 @@ extension GenericUnixToolchain {
         }
       }
 
+      if targetInfo.sdkPath != nil {
+        for libpath in targetInfo.runtimeLibraryImportPaths {
+          commandLine.appendFlag(.L)
+          commandLine.appendPath(VirtualPath.lookup(libpath.path))
+        }
+      }
+
       if !isEmbeddedEnabled && !parsedOptions.hasArgument(.nostartfiles) {
-        let swiftrtPath = VirtualPath.lookup(targetInfo.runtimeResourcePath.path)
-          .appending(
-            components: targetTriple.platformName() ?? "",
-            String(majorArchitectureName(for: targetTriple)),
-            "swiftrt.o"
-          )
-        commandLine.appendPath(swiftrtPath)
+        let rsrc: VirtualPath
+        // Prefer the swiftrt.o runtime file from the SDK if it's specified.
+        if let sdk = targetInfo.sdkPath.flatMap({ VirtualPath.lookup($0.path) }) {
```

Comment:
Nit:

```suggestion
        if let sdk = targetInfo.sdkPath, let sdk = VirtualPath.lookup(sdk.path) {
```

might be a bit more accessible?